src/utils.rs

```diff
@@ -6,3 +9,18 @@ pub fn set_threading(num_threads: usize) {
         .build_global()
         .unwrap();
 }
+
+cached_key! {
```

Comment:
Can you please give a little more information on what `cached_key!` does? Possibly add a comment. I can guess what it does, like saving loading time if you load N times the same image, but what if you don't do that? If you simply load 2 huge images and add them together, does the cache use more memory? If so, would it be possible to scan the cmdline args before doing anything else and use a cache only for the images that are loaded several time?

Reply:
I thought about it, and yes it'll keep the image in memory even when it is not needed anymore which is an issue. I'll try to find a way to make that cache system smarter by only keeping the image that are needed more than 1 time.